Core/DISET/private/Transports/M2SSLTransport.py

```diff
@@ -125,20 +126,85 @@ def initAsServer(self):
     return S_OK()
 
   def close(self):
+    # pylint: disable=line-too-long
     """ Close this socket. """
+
     if self.oSocket:
+      # TL;DR:
+      # Do NOT touch that method
+      #
       # Surprisingly (to me at least), M2Crypto does not close
-      # the socket when calling SSL.Connection.close
-      # It only does it when the garbage collector kicks in
-      # We have to manually close it here, otherwise the connections
-      # will hang forever
+      # the underlying socket when calling SSL.Connection.close
+      # It only does it when the garbage collector kicks in (see ~M2Crypto.SSL.Connection.Connection.__del__)
+      # If the socket is not closed, the connection may hang forever.
```

Comment:
Do you think it's worth a (bug?) report to M2Crypto?

Reply:
There already was but rejected https://gitlab.com/m2crypto/m2crypto/-/issues/201

Reply:
I changed my mind and re-open one, with a PR https://gitlab.com/m2crypto/m2crypto/-/merge_requests/247